Student/TechDebt/iCanvas/Router.m

```diff
@@ -1,394 +0,0 @@
-            UIViewController *returnedController = [self controllerForClass:classOrBlock params:params];
-            handler(returnedController);
-        } else {
-            UIViewController *(^blockForPath)(NSDictionary *, id) = classOrBlock;
-            UIViewController *returnedController = blockForPath(params, nil);
-            handler(returnedController);
-        }
-    }];
-}
-
-#pragma mark - Primary iPad Routing Methods
-- (UIViewController *)controllerForHandlingBlockFromURL:(NSURL *)url {
-    __block UIViewController *returnedController;
-    
-    [self matchURL:url matchHandler:^(NSDictionary *params, id classOrBlock) {
-        if (class_isMetaClass(object_getClass(classOrBlock))) { // it's a class
-            returnedController = [self controllerForClass:classOrBlock params:params];
-        } else {
-            UIViewController *(^blockForPath)(NSDictionary *, id) = classOrBlock;
-            returnedController = blockForPath(params, nil);
-        }
-    }];
-    
-    return returnedController;
-}
-
-- (UIViewController *)controllerForHandlingBlockFromViewModel:(CBIViewModel *)viewModel {
-    __block id returnedController;
-    NSURL *matchURL = [NSURL URLWithString:[viewModel.model.path realURLEncodedString]];
-
-    [self matchURL:matchURL matchHandler:^(NSDictionary *params, id classOrBlock) {
-        if (class_isMetaClass(object_getClass(classOrBlock))) { // it's a class
-            returnedController = [self controllerForClass:classOrBlock params:params];
-        } else {
-            id(^blockForPath)(NSDictionary *, id) = classOrBlock;
-            returnedController = blockForPath(params, viewModel);
-        }
-    }];
-
-    if([returnedController conformsToProtocol:@protocol(PageViewEventLoggerLegacySupportProtocol)]) {
-        NSString *decodedUrl = [matchURL.absoluteString stringByRemovingPercentEncoding];
-        id<PageViewEventLoggerLegacySupportProtocol> controller = returnedController;
-        [controller setPageViewEventName: decodedUrl];
-    }
-    
-    return returnedController;
-}
-
-- (UIViewController *)routeFromController:(UIViewController *)sourceController toURL:(NSURL *)url {
-    return [self routeFromController:sourceController toURL:url withOptions:nil];
-}
-
-- (UIViewController *)routeFromController:(UIViewController *)sourceController toURL:(NSURL *)url withOptions:(NSDictionary *)options {
-    UIViewController *destinationViewController = [self controllerForHandlingBlockFromURL:url];
-
-    if (!destinationViewController && self.fallbackHandler) {
-        self.fallbackHandler(url, sourceController);
-        return nil;
-    }
-
-    if ([destinationViewController isKindOfClass:[UIAlertController class]]) {
-        if (((UIAlertController *)destinationViewController).preferredStyle == UIAlertControllerStyleAlert) {
-            [sourceController presentViewController:destinationViewController animated:YES completion:nil];
-            return nil;
-        }
-    }
```

Comment:
Looks like this logic is being left out of the consolidation.. This allowed us to route to a `UIAlertController` with a `preferredStyle` of `.alert` and it would automatically present it. There's one case that I know of (which I'll point out in a separate comment) but there might be others. Ideas on how to handle this?

Reply:
this looks like it should be handled by `show` in the router, much like how present / show is already decided.

Reply:
Fixed in latest commit. I don't know why it doesn't also do `present` with action sheets. UIAlertController should always be presented.